src/sagemaker/workflow/emr_step.py

```diff
@@ -60,6 +63,20 @@ def to_request(self) -> RequestType:
 
         return config
 
+    def validate_prop_dicts(self):
+        """Validate k,v structure of properties dictionaries, err if not provided"""
+        if isinstance(self.properties, list) and self.properties:
+            for field_dict in self.properties:
+                try:
+                    if "key" in field_dict.keys():
+                        field_dict["Key"] = field_dict.pop("key")
+                    if "value" in field_dict.keys():
+                        field_dict["Value"] = field_dict.pop("value")
+                except KeyError:
+                    raise MissingParametersError(
```

Comment:
As you already check the keys in the `field_dict`, would it still throw a KeyError?